vm/src/builtins/float.rs

```diff
@@ -51,6 +51,14 @@ impl From<f64> for PyFloat {
     }
 }
 
+impl From<f32> for PyFloat {
+    fn from(value: f32) -> Self {
+        PyFloat {
+            value: value as f64,
+        }
+    }
+}
```

Comment:
Since Rust is strict for f32 to f64 conversion, I wish we also do same policy for RustPython codes. `impl_array_element` can take `PyFloat::from` instead of `PyFloat`. Then `f32` can be covered by `pyfloat_from_f32` instead of `PyFloat::from`